### PR TITLE
Improve Repository docs

### PR DESCRIPTION
**Pull request by mwichmann**
Give a few more details. Add example for another usecase to the User Guide.

At the same time - convert more "construction environment" and "construction variable" to entities in files already being edited.

Normalize on NEXT_RELEASE (since 4.8.1, some changed had used NEXT_RELEASE and some on NEXT_VERSION).

No code/test changes: Environment.py was a docstring change and EnvironmentTests.py comment changes.

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [X] I have updated `CHANGES.txt` and `RELEASE.txt` (and read the `README.rst`).
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: doc/user/repositories.xml [hunk 1]
```diff
@@ -94,15 +94,15 @@ int main() { printf("Hello, world!\n"); }
...
+    with the exception that &SCons; will automatically filter out
     the current directory and any non-existent
     directories from the list.
 
     </para>
 
   </section>
 
-  <section>
+  <section id="sec-repository-finding-sources">
```
Comment (bdbaddog) [inline doc/user/repositories.xml]: should this be `sect-`?

File: doc/user/repositories.xml [hunk 2]
```diff
@@ -234,17 +234,17 @@ hello.c:1: hello.h: No such file or directory
     <para>
 
     In order to inform the C compiler about the repositories,
-    &SCons; will add appropriate
-    <literal>-I</literal> flags to the compilation commands
-    for each directory in the &cv-CPPPATH; list.
-    So if we add the current directory to the
-    construction environment &cv-CPPPATH; like so:
+    &SCons; will add appropriate source file inclusion
+    directived (<literal>-I</literal> or <literal>/I</literal> flags)
```
Comment (bdbaddog) [inline doc/user/repositories.xml]: directive?

File: doc/user/repositories.xml (hunk 1 above)
Comment (mwichmann) [reply doc/user/repositories.xml]: That was the intent. Doesn't break anything this way, but I'll adjust.

File: doc/user/repositories.xml (hunk 2 above)
Comment (mwichmann) [reply doc/user/repositories.xml]: Indeed.

File: doc/user/repositories.xml [hunk 3]
```diff
@@ -472,6 +472,18 @@ int main() { printf("Hello, world!\n"); }
...
+    <para>
+
+    Note that while other files are searched through the chain of
+    repositories, &SConstruct; is special - it must be found either in
+    the current directory or the first directory
+    specified using the <literal>-Y</literal>
+    (or the <literal>--repository</literal> or
+    <literal>--srcdir</literal> synonyms)
+    command line option, or the build will abort.
+
```
Comment (mwichmann) [inline doc/user/repositories.xml:484]: I ran into this doing an experiment, and it seemed worth mentioning... we don't have to keep.

Comment (bdbaddog) [reply doc/user/repositories.xml:484]: Interesting.  I think it's quite uncommon to specify more than one repository, but definitely worth noting.
Not sure if I wouldn't consider that a bug.. but it's off on the edges so unless it breaks someones expectations.. I think it's ok for now.

**Pull request merged**